### PR TITLE
Basic server user authentication functionallity

### PR DESCRIPTION
**Pull request by bbodine10**
Created post requests for creating basic user account, that stores user in db. Created post request for user login. Run npm i dotenv bcrypt @hapi/joi jsonwebtoken to install new needed packages.

### CODE REVIEW

**Comment by dbaris** [comment]
`node_modules/*` should be added to the gitignore so we don't have to manage all those files in the repository

**Comment by bbodine10** [comment]
> `node_modules/*` should be added to the gitignore so we don't have to manage all those files in the repository

I created a new pull request with a .gitignore added that ignores node_modules/*. Should we also delete the node_modules folder.

**Comment by dbaris** [comment]
Yeah, I don't think we want it in the repository as it should just be managed locally. You can do it as part of this MR (haven't tested but I think this should work):

```
git rm -r node_modues/
git commit -m "<message"
git push -u origin <branch"
```

It's fine to add the .gitignore file in a separate MR but you can also do it as part of this one, if it's easier.

**Comment by bbodine10** [comment]
Okay I believe this pull request is updated so everything in node_modules/ is deleted. The .gitignore is being added in another  pull request, since I had already done that.

File: server/.env [hunk 1]
```diff
@@ -0,0 +1,2 @@
+ACCESS_TOKEN_SECRET=628c1f0165007a86b31d9c62ba5320d3aac1fc9363ec16f1fd8009be26f683aecadb39627e607bd80c37dc82419074575cc1234996e3e39d06494d580d9d8be
+REFRESH_TOKEN_SECRET=00ba6fad2d89be129e94235bd5ca6a123c55415b521f882d3951c2ec54924907b0b7262a53f8c042c2aee8b25946d5f8bd6c26b698dbfc83dfe167923c175548
```
Comment (dbaris) [inline server/.env:2]: We should not be keeping our secrets public on Github :) These should be managed on a system-by-system basis. You can store them locally in your `.bash_profile` so they load every time you open Terminal. Once we set the app up to run remotely, we'll need to securely configure these values for that server as well.

tldr; this file should be removed

**Comment by dbaris** [comment]
From the latest diff, it looks like all the node files are still there...

**Comment by bbodine10** [comment]
> From the latest diff, it looks like all the node files are still there...

Wait really? In the diff for me they all say they've been deleted.

**Comment by dbaris** [comment]
Ah sorry got confused by the diff showing "Deleted file", looks good to me 👍
